entrypoint.sh

```diff
@@ -978,10 +978,8 @@ appSanitize () {
   chown -R ${GITLAB_USER}:${GITLAB_USER} ${GITLAB_BUILDS_DIR}
 
   echo "Checking uploads directory permissions..."
-  chmod -R 0750 ${GITLAB_DATA_DIR}/uploads/
-  find ${GITLAB_DATA_DIR}/uploads -type f -exec chmod 0644 {} \;
-  find ${GITLAB_DATA_DIR}/uploads -type d -not -path ${GITLAB_DATA_DIR}/uploads -exec chmod 0755 {} \;
-  chown ${GITLAB_USER}:${GITLAB_USER} ${GITLAB_DATA_DIR}/uploads/
+  chmod -R u=rwX,g=rX,o= ${GITLAB_DATA_DIR}/uploads/
+  chown -R ${GITLAB_USER}:${GITLAB_USER} ${GITLAB_DATA_DIR}/uploads/
```

Comment:
Only here we do a recursive `chown` / `chmod`

The GitLab suggested permission settings, set directories to `0755` and files to `0644`. But your changes sets everything to `750`. Not sure if this could cause any problems.